### ISSUE TITLE
Export to file option

### ISSUE THREAD

**Issue by jakelowen** [okgrow/merge-graphql-schemas#122]
Just a suggestion, but it would be awesome if there were an option to take the combined typedefs export them out into a combined file. Same with resolvers.

I envisioning a project folder structure where I have distinct server, client, and schema folders.  Inside the schema folder merge-graphql-schemas could process all the types/ and resolvers/ and spit out a combined typeDefs.graphql and resolvers.js.

I'd then take those 2 files drop them into my server folder (for direct use) and client folder to aid in mocking of the server.

**Comment by cfnelson** [comment]
@jakelowen I am unsure if I fully understand your suggestion. This library executes at runtime and wasn't designed to run as a preprocessor during build time. I am having difficulty understanding this use case and why this package should write to a file.

Are you simply wishing for this library to provide a function you can call to write your schema and resolvers to a specific file path that you specify? 

As far as I understand the use case you suggested could be easily handled in your own App just by saving the returned `typeDefs` & `resolvers` using `fs`.

Or is there a specific change to this library that is required in order for you to be able to save the return response to file?

**Comment by ldgarcia** [comment]
@cfnelson, @jakelowen I am searching for a way to compile multiple GraphQL files into a single `schema.graphql` to use with [serverless-appsync-plugin](https://github.com/sid88in/serverless-appsync-plugin), for example.

**Comment by jakelowen** [comment]
Thanks @cfnelson for your thoughtful reply. I think your description of a preprocessor is exactly what I had in mind. I understand of course if preprocessing is out of scope for this project. If so, please feel free to close the issue.

If I were to speak more directly to my use case for a gql preprocessor type functionality, I would characterize it as:   "I want to use merge-graphql-schemas as designed to combine my schemas at run time for my server. However for my client/frontend I would love to have merge-graphql-schemas act as a preprocessor to compile my typedefs to one big file such that I could send it to my front end team (still me, just wearing a different hat) so they could use it for mocking. This would allow both the client and server teams to operate in parallel using methods of 'schema driven development'"

**Comment by cfnelson** [comment]
@jakelowen Thanks for your response, it's helpful to understand the reasoning and why behind the issue raised. We have done something similar in the past but, in an automated fashion for testing. We haven't produced a single schema file for human consumption before. 

Will have a think about it and think if it makes sense for it to be part of this package or maybe its own pkg.

**Comment by konsumer** [comment]
It just merges the type-defs into a string, so you can do this:

```js
import { fileLoader, mergeTypes } from 'merge-graphql-schemas'
import { writeFileSync } from 'fs'

const typeDefs = mergeTypes(fileLoader(`${__dirname}/schema/**/*.graphql`), { all: true })
writeFileSync('joined.graphql', typeDefs)
```

**Comment by jakelowen** [comment]
Thanks @konsumer - That is great.  I might humbly propose that snippet be added to the documentation and this issue be closed.

**Comment by cfnelson** [comment]
Happy to accept a PR if someone would like to make a contribution to the pkg! 🙌

**Comment by konsumer** [comment]
Seems a bit redundant, but I added it after the bit about merging types. Feel free to modify to make it clearer, or whatever.
